### PR TITLE
Add plugin: InsightA

### PR DESCRIPTION
**Pull request by HongjianTang**
# I am submitting a new Community Plugin

This plugin can extract a set of atomic notes and create MOC by note title using LLM (currently only support OpenAI).

## Repo URL

Link to my plugin: https://github.com/HongjianTang/obsidian-insighta

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
[const extractIconEl = this.addRibbonIcon('dice', `${this.manifest.name}: Extract Notes`, async (evt: MouseEvent) => {](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/main.ts#L23)
That icon does not make much sense here, also [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[if (!process.env.OPENAI_API_KEY) {](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/main.ts#L80), [const apiKey = process.env.OPENAI_API_KEY;](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/api.ts#L16)
This API is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[containerEl.createEl('h2', { text: 'Create setting' });](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/settings.ts#L54), [containerEl.createEl('h2', { text: 'LLM setting' });](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/settings.ts#L74)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

**Comment by HongjianTang** [comment]
> [const extractIconEl = this.addRibbonIcon('dice', `${this.manifest.name}: Extract Notes`, async (evt: MouseEvent) => {](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/main.ts#L23) That icon does not make much sense here, also [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> [if (!process.env.OPENAI_API_KEY) {](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/main.ts#L80), [const apiKey = process.env.OPENAI_API_KEY;](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/api.ts#L16) This API is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.
> 
> [containerEl.createEl('h2', { text: 'Create setting' });](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/settings.ts#L54), [containerEl.createEl('h2', { text: 'LLM setting' });](https://github.com/HongjianTang/obsidian-insighta/blob/4ace13241d3e3ff2bf3ec21b5f62b056e85ed8e6/src/settings.ts#L74) [Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

Hi joethei, thanks for review. I've hide the button, rename setting headers and set isDesktopOnly to true in mainfest. I thought it's ready for review again.

**Comment by HongjianTang** [comment]
Hi @joethei,

Just a gentle reminder about the PR I updated two weeks ago based on your feedback. Could you please take a moment to review it when convenient?

Thanks.

**Comment by liamcain** [comment]
- [Credits go to Liam's Periodic Notes Plugin](https://github.com/HongjianTang/obsidian-insighta/blob/8fe69eaaf0e39f3699b5933ed6ec34e6dba0e2a0/src/suggest.ts#L1) the official Suggest has been exported from the API. Please change this to use AbstractInputSuggest. It doesn't require any additional dependencies and will work more reliably within Obsidian.
- [sections = this.app.metadataCache.getFileCache(file)?.sections](https://github.com/HongjianTang/obsidian-insighta/blob/8fe69eaaf0e39f3699b5933ed6ec34e6dba0e2a0/src/view-manager.ts#L77) a warning: this information is likely going to be stale. I wouldn't suggest using this as a way to find the end of the frontmatter. You're better off using a regex search of the file contents for now until we export something better from the API

File: community-plugins.json [hunk 1]
```diff
@@ -9261,6 +9261,12 @@
         "repo": "HananoshikaYomaru/obsidian-incomplete-files"
     },
     {
+        "id": "insighta",
+        "name": "InsightA",
+        "author": "Hongjian Tang",
+        "description": "Extract a set of atomic notes from a long article and create MOC by note title using LLM",
+        "repo": "HongjianTang/obsidian-insighta"
+    }
```
Comment (liamcain) [inline community-plugins.json:9269]:
```suggestion
    },
```

File: community-plugins.json [hunk 2]
```diff
@@ -9261,6 +9261,12 @@
         "repo": "HananoshikaYomaru/obsidian-incomplete-files"
     },
     {
+        "id": "insighta",
+        "name": "InsightA",
+        "author": "Hongjian Tang",
+        "description": "Extract a set of atomic notes from a long article and create MOC by note title using LLM",
+        "repo": "HongjianTang/obsidian-insighta"
+    },
```
Comment (liamcain) [inline community-plugins.json:9269]:
```suggestion
    },
    {
```

**Comment by liamcain** [comment]
Plugin added in this commit: https://github.com/obsidianmd/obsidian-releases/commit/5c28e4eea36ae5849909be4c8366492d2de1e48b

**Pull request closed without merging**